holochain-introduction-for-developers/holochain-brief-introduction.md

```diff
@@ -0,0 +1,96 @@
+>
+> The purpose of this document is to help you quickly understand:
+>
+> * Why we're building Holochain
+> * What the project consists of
+> * What advantages it has over server-based and blockchain-based architectures
+
+## The problem
+
+The Internet is decentralized by design. Many services, such as the web and email, have no central points. But in order to move through the Internet confidently, we need **trust.** How can you know I am who I say I am, and how can you be sure that I'm acting with integrity?
+
+Centralized services act as proxies of trust. We may not be able to trust a million anonymous peers, but we can trust these big services to faithfully negotiate our interactions with those peers — **until we can't.** These systems are beginning to show weaknesses that are direct consequences of centralization.
+
+* **They are more attractive to attackers.**
+* **They leave users vulnerable to service outages.**
+* **They concentrate power in the hands of a few.**
+* **They burden the service operator with liabilities and costs.**
+
+Blockchain opened our imaginations to the possibility of trustable, decentralized coordination with untrusted peers, but suffers from efficiency problems that are inherent to its design.
+
+## Holochain: fully sovereign, distributed, trustable coordination
+
+Holochain gives you the tools to create applications that enable direct, trustable interactions among sovereign peers. It does this without needing the central coordination of a server or a group of miners. We are building:
+
+* A **peer-to-peer protocol** that gives participants the power to act collectively as an 'immune system,' pushing out bad actors.
+* An open-source **application development framework** that abstracts away complexity and allows you to focus on your core functionality.
+* A set of **core apps and utilities**, including an identity manager, a private data vault, an app store, and a software distribution platform.
+* A **distributed hosting ecosystem** that bridges Holochain to the web so people can use your apps as if they were traditional, server-based apps.
+
+### Uncompromisable sovereignty
+
+Each participant is in control of their own identity and data. Private data stays private, and public data is signed to prove ownership and authenticity.
+
+### Intrinsic data integrity
+
+Rather than ensuring trust by getting everyone to agree on a single data structure, Holochain lets the data speak for itself. Every participant has a copy of an app's validation rules, which gives them the power to audit other participants' data.
+
+### Resilience through interconnection
+
+Everyone is part of a peer-to-peer network that replicates public data, audits it, and spreads warnings of bad actors. Third-party validators are chosen at random to thwart collusion.
```

Comment:
"that replicates public data"... what about the ability to distribute private data?? I have a membrane to mark my DHT submission as private - but want the resilience (redundancy) for my data - is that to be alluded here or another feature?

Reply:
@d3sbar could you go into the use case a little more? Private data, by nature, is not distributed. I can see a use case for 'hidden in plain sight' data -- that is, data that's encrypted before it's sent to the DHT. I feel like it's a decent pattern for data that's meant for a few eyes only, less appropriate for storing completely private data (entry types that would normally be marked 'private').

Thoughts:

* Arthur recommends against this. His big concern is that all cryptography eventually gets broken, given enough time, and do you really want that data living in a DHT forever? I personally think this is more of a theoretical concern than a pragmatic one.
* If you're encrypting using a symmetric key (which you should -- more quantum resistant and much faster than public key encryption) you typically distribute the symmetric key to intended recipients by first encrypting it with their public key. This is how most public-key encryption _actually_ works behind the scenes.
* Regardless of whether you encrypt using a symmetric key or your recipients' public keys, you will never be able to revoke access to that data once you've put it out there. And this reflects reality anyway: the first time your recipient opens the encrypted data, they can save it in unencrypted form on their hard drive.
* Encrypted data can't be validated by peers because they can't decrypt it. The validation function would have to be something like `return true`. In this scenario, validating peers are reduced to just ignorant deputies. This probably isn't a problem, but it depends on your app's needs.

If the big concern is resilience for private source chain entries, I believe there will be an app for backing up private source chain data and restoring it to your device.

Reply:
@d3sbar do you feel like this sentence is ambiguous and implies that all data is public (a la blockchain)?